lib/AST/ASTImporter.cpp

```diff
@@ -1577,6 +1577,9 @@ Decl *ASTNodeImporter::VisitStaticAssertDecl(StaticAssertDecl *D) {
   DeclContext *DC = Importer.ImportContext(D->getDeclContext());
   if (!DC)
     return nullptr;
+  Decl *AlreadyImported = Importer.GetAlreadyImportedOrNull(D);
```

Comment:
Why do we need this check in the static assert visitor? Did you see an error that this resolve?

Reply:
I have a test code that triggered the assertion in `addHiddenDecl` that was fixed with changes in `VisitFunctionDecl`. When the new assertion (in `ASTImporter::Import(Decl*)`) was added the same test code triggered the new assertion, this was fixed with change in `VisitStaticAssertDecl`. This is really an indpendent new problem but it would be too much administration to have a separate issue for it (or not?). (I know about more other test code that causes the new assertion to be triggered in other places too.)